tiles/src/main/java/com/protomaps/basemap/layers/Water.java

```diff
@@ -51,25 +102,83 @@ public void processFeature(SourceFeature sf, FeatureCollector features) {
         .setMinPixelSize(3.0)
         .setBufferPixels(8);
 
+      String kind = "other";
+      String kind_detail = "";
+      var reservoir = false;
+      var alkaline = false;
+
+      // coallese values across tags to single kind value
+      if (sf.hasTag("natural", "water", "bay", "strait", "fjord")) {
+        kind = sf.getString("natural");
+        if (sf.hasTag("water", "basin", "canal", "ditch", "drain", "lake", "river", "stream")) {
+          kind_detail = sf.getString("water");
+
+          // This is a bug in Tilezen v1.9 that should be fixed in 2.0
+          // But isn't present in Protomaps v2 so let's fix it preemtively
+          if( kind_detail == "lake" ) {
+            kind = "lake";
+          }
+
+          if (sf.hasTag("water", "lagoon", "oxbow", "pond", "reservoir", "wastewater")) {
+            kind_detail = "lake";
+          }
+          if (sf.hasTag("water", "reservoir")) {
+            reservoir = true;
+          }
+          if (sf.hasTag("water", "lagoon", "salt", "salt_pool")) {
+            alkaline = true;
+          }
+        }
+      } else if (sf.hasTag("waterway", "riverbank", "dock", "canal", "river", "stream", "ditch", "drain")) {
+        kind = "water";
+        kind_detail = sf.getString("waterway");
+      } else if (sf.hasTag("landuse", "basin")) {
+        kind = sf.getString("landuse");
+      } else if (sf.hasTag("landuse", "reservoir")) {
+        kind = "water";
+        kind_detail = sf.getString("landuse");
+        reservoir = true;
+      } else if (sf.hasTag("leisure", "swimming_pool")) {
+        kind = "swimming_pool";
+      } else if (sf.hasTag("amenity", "swimming_pool")) {
+        kind = "swimming_pool";
+      }
+
+      feature.setAttr("pmap:kind", kind);
+
+      // Optional tags
+      if (kind_detail != "") {
+        feature.setAttr("pmap:kind_detail", kind_detail);
+      }
+      if (sf.hasTag("water", "reservoir") || reservoir) {
+        feature.setAttr("reservoir", true);
```

Comment:
@bdon Do you want these to be booleans or `1` int values?

Reply:
I prefer booleans, but OMG and MB and Tilezen and PM all do different things.

Given you like booleans, I believe this is a NOOP.